### PR TITLE
sp_ineachdb: support Azure SQL DB as sp_MSforeachdb drop-in

### LINKED ISSUE #3943: sp_ineachdb: support Azure SQL DB as drop-in sp_MSforeachdb replacement
## Problem

`sp_ineachdb` doesn't work on Azure SQL DB today. Two issues:

1. The execution loop builds a 3-part name dynamic call — `[dbname].sys.sp_executesql` — at [sp_ineachdb.sql:359-360](https://github.com/BrentOzarULTD/SQL-Server-First-Responder-Kit/blob/dev/sp_ineachdb.sql#L359-L360). Azure SQL DB forbids cross-database calls and 3-part names.
2. Callers of `sp_MSforeachdb` commonly pass commands like `USE [?]; SELECT ... FROM [?].sys.tables;`. Both constructs fail in Azure SQL DB.

Azure SQL Managed Instance (EngineEdition 8) supports cross-db calls and already works — this only affects Azure SQL DB (EngineEdition 5).

## Goal

Make `sp_ineachdb` a drop-in replacement for `sp_MSforeachdb` on Azure SQL DB, so that callers like `sp_Blitz` that already use `sp_MSforeachdb`-style commands work without code changes when they swap to `sp_ineachdb`.

Since Azure SQL DB has exactly one user database reachable from a session, "run in each database" collapses to "run in the current database." The procedure should:

- Detect Azure SQL DB via `SERVERPROPERTY('EngineEdition') = 5`.
- Populate `#ineachdb` with only the current database (existing filter logic still applies).
- Execute `@command` via `EXEC sys.sp_executesql` (no 3-part name).
- Rewrite the incoming `@command` to strip constructs that won't work:
  - `USE [?];` / `USE [?]` / `USE ?;` / `USE ?` → removed
  - `[?].schema.object` / `?.schema.object` → `schema.object`
- Leave all other platforms (box product, Managed Instance, Synapse) on the existing code path byte-for-byte.

## Known limitations of the proposed fix

- String-literal contents containing the placeholder get rewritten too (already a latent issue with the existing `REPLACE(@command, @replace_character, ...)` on [sp_ineachdb.sql:339](https://github.com/BrentOzarULTD/SQL-Server-First-Responder-Kit/blob/dev/sp_ineachdb.sql#L339)).
- Non-canonical whitespace like `USE  [?];` (double space) or `[?] . sys . tables` won't match the `REPLACE` patterns. These are rare in practice.

## Out of scope

- Rewriting sp_Blitz et al. to actually call `sp_ineachdb` on Azure — that's a follow-up once this lands.

PR incoming.

### PR DESCRIPTION
**Pull request by BrentOzar**
Fixes #3943

## Summary

- Detect Azure SQL DB via `SERVERPROPERTY('EngineEdition') = 5` and branch around the two things Azure forbids: 3-part names and `USE` statements.
- On Azure: seed `#ineachdb` with only the current database, execute `@cmd` with plain `EXEC sys.sp_executesql @cmd` (no 3-part dynamic call), and rewrite `@command` to strip `USE [?];` variants and collapse `[?].schema.object` / `?.schema.object` to 2-part names.
- Box SQL, Managed Instance (EngineEdition 8), Synapse, and SQL Edge keep the existing code path byte-for-byte — all changes are gated on `@IsAzureSqlDb = 1`.

This lets callers of `sp_MSforeachdb` (sp_Blitz et al.) swap in `sp_ineachdb` on Azure SQL DB without rewriting their `USE [?]; SELECT ... FROM [?].sys.xxx` command strings.

## Known limitations (documented in the issue)

- String literals containing the placeholder get rewritten too (latent issue with the existing `REPLACE(@command, @replace_character, ...)`).
- Non-canonical whitespace like `USE  [?];` or `[?] . sys . tables` won't match the `REPLACE` patterns. Rare in practice.

## Out of scope

- Updating sp_Blitz et al. to call `sp_ineachdb` on Azure — separate follow-up.

## Test plan

- [ ] Box SQL: run existing `sp_ineachdb` test calls — behavior unchanged across user/system DBs, filters, AG edge cases.
- [ ] Box SQL: confirm `@print_command_only = 1` still prints the 3-part `[db].sys.sp_executesql` form.
- [ ] Azure SQL DB: `EXEC sp_ineachdb @command = N'USE [?]; SELECT DB_NAME() AS db, COUNT(*) AS tables FROM [?].sys.tables;'` returns one row for the current DB with correct values.
- [ ] Azure SQL DB: `EXEC sp_ineachdb @command = N'SELECT ? AS placeholder;'` returns `[current_db]`.
- [ ] Azure SQL DB: `EXEC sp_ineachdb @command = N'SELECT 1;', @print_command_only = 1` prints the command without crashing.
- [ ] Azure SQL DB: confirm filter parameters (`@name_pattern`, `@user_only`, etc.) still correctly include/exclude the current DB.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: sp_ineachdb.sql [hunk 1]
```diff
@@ -162,7 +182,16 @@ BEGIN
 3)If we find a [, we begin to accumulate the result until we reach closing ], (jumping over escaped ]]).
 4)Finally, tabs, line breaks and spaces are removed from unquoted names
 */
-WITH C
+IF @IsAzureSqlDb = 1
+BEGIN
+  /* Azure SQL DB: the session is bound to one user database. Seed with it and
+     let the downstream filter DELETEs decide whether it survives. */
+  INSERT #ineachdb(id, name, is_distributor)
+  SELECT DB_ID(), DB_NAME(), 0;
```
[the remark this answers is not in the document]
Comment (BrentOzar) [reply sp_ineachdb.sql:190]: That's idiotic, Copilot.
